src/bandersnatch_filter_plugins/allowlist_name.py

```diff
@@ -176,3 +192,46 @@ def _check_match(self, name: str, version_string: str) -> bool:
                 )
                 return True
         return False
+
+
+class AllowListRequirements(AllowListRelease):
+    name = "allowlist_requirements"
+    allowlist_package_names: List[Requirement] = []
+
+    def _determine_filtered_package_requirements(self) -> List[Requirement]:
+        """
+        Parse the configuration file for
+        [allowlist]
+        requirements_path = /where_they_are
+        requirements =
+            requirements.txt
+
+        Returns
+        -------
+        list of packaging.requirements.Requirement
+            For all PEP440 package specifiers
+        """
+
+        try:
+            requirements_path = Path(self.allowlist["requirements_path"])
+        except KeyError:
+            requirements_path = Path()
+
+        try:
+            lines = self.allowlist["requirements"]
+            requirements_lines = lines.split("\n")
+        except KeyError:
+            requirements_lines = []
+
+        filtered_requirements: Set[Requirement] = set()
+
+        for requirement_line in requirements_lines:
+            requirement_line = requirement_line.strip()
+            if not requirement_line or requirement_line.startswith("#"):
+                continue
+            requirement_line, *_ = requirement_line.split('#')
```

Comment:
```suggestion
            requirement_line, _ = requirement_line.split('#', maxsplits=1)
```
Big fan of max splits

Reply:
will do